libraries/Microsoft.Bot.Connector/Authentication/JwtTokenValidation.cs

```diff
@@ -186,6 +164,22 @@ public static string GetAppIdFromClaims(IEnumerable<Claim> claims)
             return appId;
         }
 
+        /// <summary>
+        /// Validates the identity claims against the <see cref="ClaimsValidator"/> in <see cref="AuthenticationConfiguration"/> if present. 
+        /// </summary>
+        /// <param name="authConfig">An <see cref="AuthenticationConfiguration"/> instance.</param>
+        /// <param name="claims">The list of claims to validate.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        /// <exception cref="UnauthorizedAccessException">If the validation returns false.</exception>
+        internal static async Task ValidateClaimsAsync(AuthenticationConfiguration authConfig, IEnumerable<Claim> claims)
+        {
+            if (authConfig.ClaimsValidator != null && !await authConfig.ClaimsValidator.ValidateClaimsAsync(new List<Claim>(claims)).ConfigureAwait(false))
```

Comment:
Defining a method to receive IEnumerable to only create a list means we iterate the enumerable and then create a new list with the elements, both of which are unnecessary. I'd make this receive List or IList, or otherwise have ClaimsValidator.ValidateClaimsAsync() to receive IEnumerable

Reply:
Changed this and the method signature using a soft cast based on this article: https://blog.jetbrains.com/dotnet/2017/09/05/ienumerable-inspections-quick-fixes-resharper-rider/